edi/models/edi_backend.py

```diff
@@ -127,3 +127,62 @@ def _validate_generate_output(self, exchange_record, force=False):
     def _generate_output(self, exchange_record, **kw):
         """To be implemented"""
         raise NotImplementedError()
+
+    def exchange_send(self, exchange_record):
+        """Send exchange file."""
+        self.ensure_one()
+        exchange_record.ensure_one()
+        if exchange_record.direction != "output":
+            raise exceptions.UserError(
+                _("Record ID=%d is not meant to be sended") % exchange_record.id
+            )
+        if not exchange_record.exchange_file:
+            raise exceptions.UserError(
+                _("Record ID=%d has no file to send!") % exchange_record.id
+            )
+        # In case already sent: skip sending and check the state
+        check = self._exchange_output_check(exchange_record)
+        if not check:
+            return False
+        try:
+            self._exchange_send(exchange_record)
+        except Exception as err:
```

Comment:
this catchall exception can be dangerous as it can swallow important exceptions.
I would rely on a ctx key to raise the error on demand (eg: handy for tests or for manual test via UI).
Something like:
```suggestion
        except Exception as err:
            if self.env.context.get("_edi_send_break_on_error"):
                raise
```